flight/Modules/Actuator/actuator.c

```diff
@@ -191,76 +214,34 @@ static void actuatorTask(void* parameters)
 	{
 		PIOS_WDG_UpdateFlag(PIOS_WDG_ACTUATOR);
 
-		// Wait until the ActuatorDesired object is updated
-		bool rc = PIOS_Queue_Receive(queue, &ev, FAILSAFE_TIMEOUT_MS);
+		bool rc=false;
 
-		/* Process settings updated events even in timeout case so we always act on the latest settings */
-		if (actuator_settings_updated) {
-			actuator_settings_updated = false;
-			ActuatorSettingsGet (&actuatorSettings);
-			actuator_update_rate_if_changed (&actuatorSettings, false);
+		// If it's our first iteration, fall down to getting the
+		// config and enabling failsafe.
+		if (!first) {
```

Comment:
All these settings objects are now global, so the ```settings_updated``` variable can trigger a function update such as

```
settingsUpdate()
{
	ActuatorSettingsGet(&actuatorSettings);
	actuator_update_rate_if_changed(false);
	MixerSettingsGet(&mixerSettings);
}
```

and then ```settingsUpdate``` could be called before the loop begins, as well as inside the loop, right?

Or am I missing something?